llvm/include/llvm/BinaryFormat/DXContainer.h

```diff
@@ -541,6 +548,21 @@ struct ProgramSignatureElement {
 static_assert(sizeof(ProgramSignatureElement) == 32,
               "ProgramSignatureElement is misaligned");
 
+struct RootSignatureValidations {
+
+  static Expected<uint32_t> validateRootFlag(uint32_t Flags) {
+    if ((Flags & ~0x80000fff) != 0)
+      return llvm::make_error<BinaryStreamError>("Invalid flag");
+    return Flags;
+  }
+
+  static Expected<uint32_t> validateVersion(uint32_t Version) {
+    if (Version < 1 || Version > 2)
+      return llvm::make_error<BinaryStreamError>("Invalid Version");
```

Comment:
What version is invalid?

```suggestion
      return llvm::make_error<BinaryStreamError>("Invalid Root Signature Version");
```